### PR TITLE
feat: add LeetCode problem 61

### PR DESCRIPTION
**Pull request by ashikjhonson**
Rotate List

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: leetcode/src/61.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+* Definition for singly-linked list.
+* struct ListNode {
+* int val;
+* struct ListNode *next;
+* };
+*/
+
+struct ListNode* rotateRight(struct ListNode* head, int k){
+struct ListNode *ptr = head, *prev =NULL;
+int count = 0;
```
Comment (Panquesito7) [inline leetcode/src/61.c:23]: This code should be in a function to have it more structured, IMO.

File: leetcode/src/61.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        prev = ptr;
+        ptr = ptr->next;
+    }
+    ptr->next = head;
+    head = ptr;
+    prev->next = NULL;
+}
+return head;
+
+}
```
Comment (Panquesito7) [inline leetcode/src/61.c:47]: Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Review by Panquesito7 — DISMISSED** [review]
Thanks! 🚀

File: leetcode/src/61.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    if(count!=0)
+        k = k%count;
+    while(k--) {
+        ptr = head;
+        if(ptr==NULL)
+            return NULL;
+        if(ptr->next==NULL)
+            return ptr;
+        while(ptr->next) {
```
Comment (alexpantyukhin) [inline leetcode/src/61.c:38]: Are you sure that the Runtime complexity is O(n + k)? I see the outer loop is O(k) and the inner loop goes throught all the list so, If I'm right it's the Runtime O(n * k), no?

Comment (CascadingCascade) [reply leetcode/src/61.c:38]: > Are you sure that the Runtime complexity is O(n + k)? I see the outer loop is O(k) and the inner loop goes throught all the list so, If I'm right it's the Runtime O(n * k), no?

I think you are right that it's not O(n + k), but I dissent that the outer loop is really O(k mod n), so O(n * k) would be the worst case time complexity (When k = n - 1).

**Review by CascadingCascade — COMMENTED** [review]
I would stand by my earlier comment at the closed PR that your solution is not exactly an efficient one. If we keep track of three nodes: node at index `k % count` (pk), node at index `k % count - 1` (pk1) and list tail (tail), we can rotate the list by simply `tail.next = head; pk1.next = NULL; return pk;`. This way we get to solve the problem consistently in O(n) because everything except for the first loop only takes constant time.

**Pull request closed without merging**